api/client.go

```diff
@@ -28,17 +29,35 @@ const (
 	InfraConnectorRole = "connector"
 )
 
-type Client struct {
-	Name      string
-	Version   string
-	URL       string
-	AccessKey string
-	HTTP      http.Client
+type ClientOptions struct {
+	Name    string
+	Version string
+
+	// HTTP is the backend HTTP client. An uninitialized value will use http.DefaultClient.
+	HTTP http.Client
+
 	// Headers are HTTP headers that will be added to every request made by the Client.
 	Headers http.Header
+
 	// OnUnauthorized is a callback hook for the client to get notified of a 401 Unauthorized response to any query.
 	// This is useful as clients often need to discard expired access keys.
 	OnUnauthorized func()
+
+	ObserveFunc func(time.Time, *http.Request, *http.Response)
+}
+
+type Client struct {
+	URL       string
+	AccessKey string
+	Options   *ClientOptions
+}
+
+func NewClient(serverURL, serverAccessKey string, options ClientOptions) *Client {
```

Comment:
What's the distinction between options and these other two fields? Isn't access key just as much of an option as the rest?  You can make requests to some endpoints without an access key.

We frequently use option structs for building complex types, but in this case since the type `Client` doesn't have any state, the `Client` type itself is basically an options struct.

Reply:
I was thinking URL and AccessKey were mandatory fields while `Options` contains optional fields but that's doesn't look to be the case. Will roll this back